bundles/org.openhab.binding.neohub/src/main/java/org/openhab/binding/neohub/internal/NeoHubHandler.java

```diff
@@ -179,25 +179,64 @@ public synchronized NeoHubReturnResult toNeoHubSendChannelValue(String commandSt
      * 
      */
     protected NeoHubInfoResponse fromNeoHubFetchPollingResponse() {
-        if (socket == null || config == null) {
-            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.NONE);
+        if (socket == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
+            logger.warn(MSG_HUB_COMM);
+            synchronized (this) {
+                cachedInfoResponse = null;
+            }
```

Comment:
Synchronization serves no purpose here since assigning a variable is an atomic operation. Please remove the synchronized block. If you really want to you can make cachedInfoResponse volatile.

Reply:
Oh that’s interesting. I come from a background in other compiled languages, where pointer assignments are not guaranteed to be atomic. But if Java has mechanisms to ensure atomicity, then I am very happy. I learned something new today.

Reply:
Well adding volatile truly makes it atomic, but I think the code here would work just fine without volatile.

Reply:
After careful thought, I realised that cachedInfoResponse is not actually needed, so I have re-coded to eliminate it entirely.